common/helpers/field/set-optional-label.js

```diff
@@ -0,0 +1,31 @@
+const { cloneDeep, get, set, flatten } = require('lodash')
+
+const i18n = require('../../../config/i18n')
+
+function setOptionalLabel([key, field]) {
+  const translated = cloneDeep(field)
+  const validations = flatten([translated.validate])
+  const isOptional =
+    validations.filter(v => v?.type === 'required' || v === 'required').length <
+    1
+  const labelPaths = [
+    'label.text',
+    'label.html',
+    'fieldset.legend.text',
+    'fieldset.legend.html',
+  ]
+
+  if (isOptional) {
+    labelPaths.forEach(path => {
+      const key = get(translated, path)
+
+      if (key) {
+        set(translated, path, `${key} (${i18n.t('optional')})`)
```

Comment:
I think this is more than good enough for now. 

If we do need the ability to specify different strings or (even none at all), it would be easy enough to do something like:

```js
const defautOptional = i18n.t('optional')
const optional = i18n.t(`${path}_optional`, defaultOptional)
set(translated, path, `${key}${optional}`)
```

To that end, might be worth putting the space and parentheses in the optional text itself...